README.md

```diff
@@ -177,9 +182,68 @@ The data in the 'commcare_locations' table comes from the Location API
 endpoint along with some additional columns from the Location Type API
 endpoint.
 
-Note that the table names 'commcare_users' and 'commcare_locations' are
-treated as reserved names and the export tool will produce an error if
-given a query specification that writes to either of them.
+Specifying the --with-organization option, which only works with SQL database
+output, will create a set of additional tables and views. First, it will create
+a view named 'commcare_location_hierarchy' that contains some of the columns
+of the 'commcare_locations' table and the location and location type names of
+parents named as follows:
+
+Column                       | Type | Note
+------                       | ---- | ----
+id                           | Text |
+location_id                  | Text | Primary key
+location_type                | Text |
+parent                       | Text |
+resource_uri                 | Text |
+location_type_name           | Text |
+loc_level1                   | Text | Same as location_id
+loc_name_level1              | Text | Same as location_type_name
+loc_level2                   | Text | Parent's location_id
+loc_name_level2              | Text | Parent's location_type_name
+loc_level3                   | Text | Grandparent's location_id
+loc_name_level3              | Text | Grandparent's location_type_name
+4 ... 7                      | Text |
+loc_level8                   | Text | 7 x parent's commcare_location_id
+loc_name_level8              | Text | 7 x parent's commcare_location_type_name
+
+Note that when exporting to a MySQL database, 'commcare_location_hierarchy'
+will be table rather than a view. In addition, each table exported by an
+Excel query specification will have a 'commcare_userid' column added so
+that it can be joined with user and location tables and a view will be
+created that performs such a join. For example, if the Excel query emits
+a table 'my_form' with columns 'id' and 'date', the created view will be
+named 'my_form_with_organization' and it will have the following columns:
```

Comment:
I had asked
> What happens to the views if a new column is added to a referenced table or an existing column's datatype changes?

@solleks responded
> They will break if we change fields in the internal users or locations tables though.

That didn't quite answer my question. What if there are changes to a _form_ table (for example, column data type changed on `my_form_with_organization`)? I believe this could happen each time the DET is run against an already-existing SQL database when there are new form submissions to be exported.

Reply:
In that case, the view will prevent the change in column data type. I presume that sqlalchemy would raise an exception at that point and the export would fail. The natural way to handle it would be to drop and recreate the view, but that requires that we maintain views even on export runs where the user doesn't specify --with-organization, which seems pretty bad.